src/main/java/io/anserini/rerank/lib/RocchioReranker.java

```diff
@@ -0,0 +1,212 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.rerank.lib;
+
+import io.anserini.analysis.AnalyzerUtils;
+import io.anserini.index.IndexReaderUtils;
+import io.anserini.rerank.Reranker;
+import io.anserini.rerank.RerankerContext;
+import io.anserini.rerank.ScoredDocuments;
+import io.anserini.util.FeatureVector;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.index.Terms;
+import org.apache.lucene.index.TermsEnum;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.BoostQuery;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.util.BytesRef;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import static io.anserini.search.SearchCollection.BREAK_SCORE_TIES_BY_DOCID;
+import static io.anserini.search.SearchCollection.BREAK_SCORE_TIES_BY_TWEETID;
+
+public class RocchioReranker implements Reranker {
+  private static final Logger LOG = LogManager.getLogger(RocchioReranker.class);
+
+  private final Analyzer analyzer;
+  private final String field;
+
+  private final int fbTerms;
+  private final int fbDocs;
+  private final float alpha;
+  private final float beta;
+  private final boolean outputQuery;
+
+  public RocchioReranker(Analyzer analyzer, String field, int fbTerms, int fbDocs, float alpha, float beta, boolean outputQuery) {
+    this.analyzer = analyzer;
+    this.field = field;
+    this.fbTerms = fbTerms;
+    this.fbDocs = fbDocs;
+    this.alpha = alpha;
+    this.beta = beta;
+    this.outputQuery = outputQuery;
+  }
+
+  @Override
+  public ScoredDocuments rerank(ScoredDocuments docs, RerankerContext context) {
+    assert(docs.documents.length == docs.scores.length);
+
+    IndexSearcher searcher = context.getIndexSearcher();
+    IndexReader reader = searcher.getIndexReader();
+
+    FeatureVector qfv = FeatureVector.fromTerms(AnalyzerUtils.analyze(analyzer, context.getQueryText()));
+
+    FeatureVector rm = computeMeanOfDocumentVectors(docs, reader);
+
+    // Rocchio weight = alpha * original query + beta * mean(top n doc embedding)
+    // Here beta = 1-alpha
```

Comment:
"Here beta = 1-alpha" -> this comment is outdated, because we're reading beta from a param now.